### DISCUSSION TITLE
How to generate static sitemap and rss of a static blog

### DISCUSSION THREAD

**Discussion by aralroca** [vercel/next.js#12209] [Help] [answered]
Recently I migrate my blog to Next.js and first of all I want to say how amazing is the new SSG methods! Now my blog is fully static!

However, I want to generate also the sitemap and rss based in my markdown files. But I achived only with `getServerSideProps`... 

Is there some way to achive this using `getStaticProps`?

**Comment by aralroca** [comment]
@Timer @timneutkens  Could it be a good idea to use something like this inside the `getStaticProps` of my `/blog` page?

```js
fs.writeFileSync('public/rss.xml', rssData)
```

I try it and it works... 🤷‍♂️

**Comment by valse** [comment]
Hi, I made an API function to do it: `pages/api/sitemap.xml.js`
that you can call by the url:
`https://.../api/sitemap.xml`
or you could also make a rewrite url as you like.

In the response I set also the correct content type and cache control header to let the CDN to proxy it for some time.

**Comment by leerob** [comment, marked as answer]
Edit: Moved to docs https://github.com/vercel/next.js/pull/28346

**Reply by aralroca** [reply]
Ok! Then using writeFileSync to public is a good idea? I finally did a similar implementation, however I wasn't sure about if It was a good idea.

**Reply by aralroca** [reply]
Ok, I see that you are creating an extra build step. I implemented this `fs.writeFileSync` inside an `getStaticProps` method...

**Reply by aralroca** [reply]
I finally did it with your way https://github.com/aralroca/aralroca.com/pull/42! Thanks a lot

**Reply by leerob** [reply]
Awesome, glad I could help!

**Comment by leerob** [comment]
We are adding [official docs on sitemaps](https://github.com/vercel/next.js/pull/28346), please leave any feedback there! 🙏
